.github/workflows/ci.yml

```diff
@@ -0,0 +1,80 @@
+name: CI
+
+# Trigger the workflow on all pushes
+on:
+  push:
+    branches:
+      - main
+
+concurrency:
+  group: ${{ github.head_ref || github.run_id }}
+  cancel-in-progress: true
+
+jobs:
+  packages:
+    name: Push package to local cache
+    runs-on: self-hosted
+    strategy:
+      matrix:
+        ghc-version: [ "ghc9101", "ghc982", "ghc964" ]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Build & test
+        run: |
+          # The -L flag outputs build logs to stderr
+          nix build -L .#${{ matrix.ghc-version }}
+
+      - name: Push package to cache
+        env:
+          ATTIC_TOKEN: ${{ secrets.ATTIC_SECRET }}
+        run: |
+          attic login --set-default public http://192.168.102.136:9200/ "$ATTIC_TOKEN"
+          attic push public $(nix path-info .#${{ matrix.ghc-version }})
+
+      - name: Push package to cachix
+        # if: github.ref == 'refs/heads/main'
+        env:
+          CACHIX_TOKEN: ${{ secrets.CACHIX_SECRET }}
+        run: |
+          cachix authtoken $CACHIX_TOKEN
+          nix path-info .#${{ matrix.ghc-version }} | cachix push clash-lang
+
+  devshells:
+    name: Push Nix developer shell to local cache
+    runs-on: self-hosted
+    strategy:
+      matrix:
+        ghc-version: [ "ghc9101", "ghc982", "ghc964" ]
+    steps:
+        # There's no need to configure Nix, the dockerfile handling the GHA has it done for us!
+        # If a dependencies are already cached, we can simply re-use them!
+
+      - name: Checkout
+        uses: actions/checkout@v4
+
+      - name: Build devshell
+        run: |
+          # Since the server is x86_64-linux, we can only build the devshell
+          # for that
+          nix build .#devShells.x86_64-linux.${{ matrix.ghc-version }}-full
+
+        # Pushes the binaries to the local network
+        # url: http://192.168.102.136:9200/public
+        # public key: public:PGGlJMx1gGmU069blMqve8tS1ndzBuriUAwGBHGOo4g=
+      - name: Push devshell to cache
+        env:
+          ATTIC_TOKEN: ${{ secrets.ATTIC_SECRET }}
+        run: |
+          attic login --set-default public http://192.168.102.136:9200/ "$ATTIC_TOKEN"
```

Comment:
This way, the secret shows up in the process list if you time it right. While the risk of leakage is small, it is poor form. Does `attic` also accept a token from stdin? Because you might be able to do
```bash
attic login --set-default public http://192.168.102.136:9200/ - <<<"$ATTIC_TOKEN"
```
which, as far as I know, does not leak the token via the process list. My assumption was that a `-` argument would be interpreted as "take it from stdin", this is pretty standard but not universal.

Reply:
Ah it also seems possible without relying on bash features:
```bash
printenv ATTIC_TOKEN | attic login --set-default public http://192.168.102.136:9200/ -
```